File: src/plugin/combat/magic/modern/god/StormOfArmadylSpellPlugin.java
package plugin.combat.magic.modern.god;

import org.redrune.game.content.combat.player.style.MagicCombatStyle;
import org.redrune.game.entity.actor.Actor;
import org.redrune.game.entity.actor.mask.Graphics;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.plugin.combat.spell.type.CombatSpellPlugin;
import org.redrune.game.entity.projectile.ProjectileManager;
import org.redrune.utility.constants.SkillConstants;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 8/7/2017
 */
public class StormOfArmadylSpellPlugin extends CombatSpellPlugin {
	
	private static final Graphics GRAPHICS = new Graphics(457);
	
	@Override
	public int delay(Player player) {
		return player.getEquipment().getWeaponId() == 21777 ? 4 : 5;
	}
	
	@Override
	public int animationId() {
		return 10546;
	}
	
	@Override
	public int hitGfx() {
		return 1019;
	}
	
	@Override
	public int maxHit(Player player, Actor target) {
		return 160 + minimumHit(player);
	}
	
	@Override
	public void cast(Player source, Actor target, MagicCombatStyle style) {
		source.setNextGraphics(GRAPHICS);
		ProjectileManager.sendProjectile(ProjectileManager.createSpeedDefinedProjectile(source, target, 1019, 30, 26, 52, 0, 0));
		if (target.isPlayer()) {
			target.toPlayer().getSkills().set(DEFENCE, target.toPlayer().getSkills().getLevel(DEFENCE) - 1);
		}
		style.sendSpell(source, target, this);
	}
	
	@Override
	public int minimumHit(Player player) {
		int start = 77;
		int level = player.getSkills().getLevelForXp(SkillConstants.MAGIC);
		int difference = level - start;
		return (difference / 2) * 10;
	}
	
	@Override
	public int spellId() {
		return 99;
	}
	
	@Override
	public double exp() {
		return 70;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.REGULAR;
	}
}

File: src/main/java/plugin/interaction/combat/magic/modern/god/StormOfArmadylSpellPlugin.kt
package plugin.interaction.combat.magic.modern.god

import org.redrune.game.content.entity.actor.combat.player.style.MagicCombatStyle
import org.redrune.game.content.plugin.combat.spell.type.CombatSpellPlugin
import org.redrune.game.entity.actor.Actor
import org.redrune.game.entity.actor.mask.Graphics
import org.redrune.game.entity.actor.player.Player
import org.redrune.game.entity.projectile.ProjectileManager
import org.redrune.utility.constants.MagicConstants.MagicBook
import org.redrune.utility.constants.SkillConstants

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 8/7/2017
 */
class StormOfArmadylSpellPlugin : CombatSpellPlugin {
    override fun delay(player: Player): Int {
        return if (player.equipment.weaponId == 21777) 4 else 5
    }

    override fun animationId(): Int {
        return 10546
    }

    override fun hitGfx(): Int {
        return 1019
    }

    override fun maxHit(player: Player, target: Actor): Int {
        return 160 + minimumHit(player)
    }

    override fun cast(source: Player, target: Actor, style: MagicCombatStyle) {
        source.setNextGraphics(GRAPHICS)
        ProjectileManager.sendProjectile(
            ProjectileManager.createSpeedDefinedProjectile(
                source,
                target,
                1019,
                30,
                26,
                52,
                0,
                0
            )
        )
        if (target.isPlayer) {
            target.toPlayer().skills[SkillConstants.DEFENCE] =
                target.toPlayer().skills.getLevel(SkillConstants.DEFENCE) - 1
        }
        style.sendSpell(source, target, this)
    }

    override fun minimumHit(player: Player): Int {
        val start = 77
        val level = player.skills.getLevelForXp(SkillConstants.MAGIC)
        val difference = level - start
        return difference / 2 * 10
    }

    override fun spellId(): Int {
        return 99
    }

    override fun exp(): Double {
        return 70.0
    }

    override fun book(): MagicBook {
        return MagicBook.REGULAR
    }

    companion object {
        private val GRAPHICS = Graphics(457)
    }
}